crates/vm/levm/src/gas_cost.rs

```diff
@@ -775,34 +775,47 @@ pub fn staticcall(
     calculate_cost_and_gas_limit_call(true, gas_from_stack, gas_left, call_gas_costs, 0)
 }
 
-pub fn fake_exponential(factor: U256, numerator: U256, denominator: U256) -> Result<U256, VMError> {
-    let mut i = U256::one();
+pub fn fake_exponential(factor: U256, numerator: U256, denominator: u64) -> Result<U256, VMError> {
+    if denominator == 0 {
+        return Err(InternalError::DivisionByZero.into());
+    }
+
+    if numerator.is_zero() {
+        return Ok(factor);
+    }
+
+    let mut i: u64 = 1;
     let mut output: U256 = U256::zero();
+    let denominator_u256: U256 = denominator.into();
 
     // Initial multiplication: factor * denominator
     let mut numerator_accum = factor
-        .checked_mul(denominator)
+        .checked_mul(denominator_u256)
         .ok_or(InternalError::Overflow)?;
```

Comment:
Maybe just multiply to a U512 if there is a real possibility of the value not fitting.

Reply:
i prefer not to, the usual case is probably not overflowing and probably u512 are more expensive themselves than a checked mul on u256